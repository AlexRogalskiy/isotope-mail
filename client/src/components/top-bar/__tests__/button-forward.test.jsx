import React from 'react';
import {shallow} from 'enzyme/build/index';
import ButtonForward from '../button-forward';

describe('ButtonForward component test suite', () => {
  test('Snapshot render, outboxEmpty, should render button', () => {
    // Given
    const props = {outboxEmpty: true, forwardMessage: jest.fn()};
    // When
    const button = shallow(<ButtonForward {...props}/>);
    // Then
    expect(button).toMatchSnapshot();
  });
  test('Snapshot render, not outboxEmpty, should NOT render button', () => {
    // Given
    const props = {collapsed: false, forwardMessage: jest.fn()};
    // When
    const button = shallow(<ButtonForward {...props}/>);
    // Then
    expect(button).toMatchSnapshot();
  });
  test('click, should trigger function', () => {
    // Given
    const props = {outboxEmpty: true, forwardMessage: jest.fn()};
    const button = shallow(<ButtonForward {...props}/>);
    // When
    button.find('button').simulate('click');
    // Then
    expect(props.forwardMessage).toHaveBeenCalledTimes(1);
  });
});
