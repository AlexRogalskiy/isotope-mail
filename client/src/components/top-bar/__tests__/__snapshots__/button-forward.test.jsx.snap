// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonForward component test suite Snapshot render, not outboxEmpty, should NOT render button 1`] = `""`;

exports[`ButtonForward component test suite Snapshot render, outboxEmpty, should render button 1`] = `
<button
  className="material-icons mdc-top-app-bar__action-item"
  onClick={[MockFunction]}
>
  forward
</button>
`;
